packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -136,11 +246,19 @@ contract DowntimeSlasher is SlasherUtil {
     address[] memory groupElectionGreaters,
     uint256[] memory groupElectionIndices
   ) public {
-    address validator = getAccounts().signerToAccount(
-      validatorSignerAddressFromSet(startSignerIndex, startBlock)
+    uint256 startBlock = startBlocks[0];
+    uint256 endBlock = endBlocks[endBlocks.length.sub(1)];
+    require(
+      endBlock.sub(startBlock).add(1) >= slashableDowntime,
+      "the provided intervals must span slashableDowntime blocks"
     );
-    checkIfAlreadySlashed(validator, startBlock);
-    require(isDown(startBlock, startSignerIndex, endSignerIndex), "Not down");
+    address validator = getValidatorAccountFromSignerIndex(signerIndices[0], startBlock);
+    require(
+      startBlock > lastSlashedBlock[validator],
+      "cannot slash validator for downtime for which they may already have been slashed"
+    );
+    require(wasDownForIntervals(startBlocks, endBlocks, signerIndices), "not down");
```

Comment:
It depends on how big were the intervals checked before.
We basically pre-compute as different actions every interval, which also have a few optimisations, so I remember that we could almost cover the 8hrs window. So, let's say that you made 6hrs intervals just to be sure.
That `wasDownForIntervals` basically check those precompute intervals, and that's not expensive, so this should support an order of months or even years.

The only true limitation today is that we use the parentSealBitmap to compute those intervals, and we only keep track of the last 4 epochs. So if you want to slash with a `slashableDowntimes > 4 epochs`, the user should pre-compute every interval without knowing if the validator is going to be down, and almost "betting" that the user will stay down.

Reply:
Made some numbers.
Intervals of 5k blocks work great. And the intervals verifier for 1000 intervals, it's also works like a charm, and consumes ~6.5M, si it could be even more.
So basically could work for even more than 5M blocks for slashableDowntime